lib/pdf_fill/extras_generator_v2.rb

```diff
@@ -14,44 +76,29 @@ def initialize(form_name: nil, submit_date: nil, start_page: 1, sections: nil)
       @questions = {}
     end
 
-    def render_question(pdf, question)
-      sorted_subquestions = question[:subquestions].sort_by do |subq|
-        metadata = subq[:metadata]
-        [metadata[:i] || 99_999, metadata[:question_suffix] || '', metadata[:question_text] || '']
-      end
-      sorted_subquestions.each do |subq|
-        value = subq[:value]
-        metadata = subq[:metadata]
-        pdf.move_down(10)
-        prefix = metadata[:question_num].to_s
-        prefix += metadata[:question_suffix] if metadata[:question_suffix].present?
-        prefix = "#{prefix}. #{metadata[:question_text].humanize}"
-        i = metadata[:i]
-        prefix += " Line #{i + 1}" if i.present?
-
-        pdf.text("#{prefix}:", { style: :normal })
-        pdf.text(value.to_s, { style: :bold })
-      end
-    end
-
     def add_text(value, metadata)
       question_num = metadata[:question_num]
-      question = (@questions[question_num] ||= { subquestions: [], overflow: false })
-      question[:subquestions] << { value:, metadata: }
-      question[:overflow] ||= metadata.fetch(:overflow, true)
+      if @questions[question_num].blank?
+        @questions[question_num] = (metadata[:i].blank? ? Question : ListQuestion).new(metadata)
+      end
+      @questions[question_num].add_text(value, metadata)
     end
 
     def populate_section_indices!
       return if @sections.blank?
 
       @questions.each do |num, question|
-        question[:section_index] = @sections.index { |sec| sec[:question_nums].include?(num) }
+        question.section_index = @sections.index { |sec| sec[:question_nums].include?(num) }
```

Comment:
Should you put a `next if question.nil?` before the `question.section_index =`

Reply:
`@questions` is a hash from question number to Question object, so when doing `.each` the latter should never be nil. In fact, there's an existing test for this method that sets up a bunch of [questions with non-contiguous numbers](https://github.com/department-of-veterans-affairs/vets-api/blob/cf58080/spec/lib/pdf_fill/extras_generator_v2_spec.rb#L28).